osu.Game/Online/SocketClientConnector.cs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.TypeExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Logging;
+using osu.Game.Online.API;
+
+namespace osu.Game.Online
+{
+    public abstract class SocketClientConnector : Component
```

Comment:
I'm not sure about calling this a `Socket` connector. That's a very specific network term that isn't necessarily applicable to usages of this. How about something like `PersistentEndpointClient` / `PersistentEndpointClientConnector`? Open to suggestion if that feels too long.